shared/controlflow/codeql/controlflow/Cfg.qll

```diff
@@ -1192,6 +1193,78 @@ module Make<LocationSig Location, InputSig<Location> Input> {
     }
   }
 
+  /** Provides the input to `ViewCfgQuery`. */
+  signature module ViewCfgQueryInputSig<FileSig File> {
+    /** The source file selected in the IDE. Should be an `external` predicate. */
+    string selectedSourceFile();
+
+    /** The source line selected in the IDE. Should be an `external` predicate. */
+    string selectedSourceLine();
```

Comment:
Question: is this predicate really needed in the signature. Can't we simply define `external` predicates in the shared library?

Reply:
External predicates can't be defined in (parameterized) modules.